crates/runtime/src/auth.rs

```diff
@@ -29,7 +30,15 @@ pub trait AuthProvider {
     fn new(auth: &Auth) -> Self
     where
         Self: Sized;
-    fn get_token(&self) -> String;
+    fn get_token(&self) -> String {
+        String::new()
+    }
+    fn get_username(&self) -> String {
+        String::new()
+    }
+    fn get_password(&self) -> String {
+        String::new()
+    }
```

Comment:
These should return Option<String> and return None as the default implementation.

Also - why do we have get_token and get_password? This feels like the abstraction is breaking down. We could still just keep `get_token` if we encoded the username/password for the flight client like `{username}:{password}` right?

Reply:
Spoke offline, we will revisit the auth interface